src/AvaloniaEdit/Rendering/TextView.cs

```diff
@@ -890,7 +890,7 @@ protected override Size MeasureOverride(Size availableSize)
             {
                 // no document -> create empty list of lines
                 _allVisualLines = new List<VisualLine>();
-                _visibleVisualLines = new ReadOnlyCollection<VisualLine>(_allVisualLines.ToArray());
+                _visibleVisualLines = new ReadOnlyCollection<VisualLine>(_allVisualLines);
```

Comment:
@Gillibald be careful with this change. I think we need to maintain generating a new collection. I already fixed that in [this commit](https://github.com/AvaloniaUI/AvaloniaEdit/commit/d5c80250404a576115df2c82a414102ba2796cdd).

See upstream project implementation:
- https://github.com/icsharpcode/AvalonEdit/blob/b6fe189e7be4a8b650c0c670fdac8b839800d9cc/ICSharpCode.AvalonEdit/Rendering/TextView.cs#L927
- https://github.com/icsharpcode/AvalonEdit/blob/b6fe189e7be4a8b650c0c670fdac8b839800d9cc/ICSharpCode.AvalonEdit/Rendering/TextView.cs#L1025